### PR TITLE
Add Past Reports List View

### PR DESCRIPTION
**Pull request by justinkonecny**
Adds the past reports list view:
 - Shows dropdown select list of all schools the current user 
 - After selecting a school, the list of reports is shown

 

Screen Shot 2021-07-01 at 7 01 26 PM

Screen Shot 2021-07-01 at 7 01 52 PM

Screen Shot 2021-07-01 at 7 02 23 PM

Screen Shot 2021-07-01 at 7 02 28 PM

### CODE REVIEW

File: src/containers/pastSubmissionsReports/PastSubmissionReports.tsx [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+      key: 'id',
+      sorter: {
+        compare: (a, b) => a.updatedAt.localeCompare(b.updatedAt),
+        multiple: 1,
+      },
+    },
+    {
+      title: 'Library Status',
+      dataIndex: 'libraryStatus',
+      key: 'id',
```
Comment (wwohlers) [inline src/containers/pastSubmissionsReports/PastSubmissionReports.tsx:56]: Why are the keys for all these columns `id`? Is that intentional?

File: src/containers/pastSubmissionsReports/PastSubmissionReports.tsx [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+      dataIndex: 'schoolId',
+      key: 'id',
+    },
+    {
+      title: 'User ID',
+      dataIndex: 'userId',
+      key: 'id',
+      sorter: {
+        compare: (a, b) =>
+          a.userId.toString().localeCompare(b.userId.toString()),
```
Comment (wwohlers) [inline src/containers/pastSubmissionsReports/PastSubmissionReports.tsx:73]: userId is an integer right? Isn't it better to just compare with `>`?

**Review by wwohlers — CHANGES_REQUESTED** [review]
Just some small things

**Review by rymaju — APPROVED** [review]
LGTM 💯

File: src/containers/pastSubmissionsReports/PastSubmissionReports.tsx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    }
+  }, [schoolId, dispatch, history]);
+
+  const availableReports: AsyncRequest<
+    ReportGenericListResponse,
+    any
+  > = useSelector(
+    (state: C4CState) =>
+      state.pastSubmissionReportsState.pastSubmissionsReports,
+  );
```
Comment (maxsebso) [inline src/containers/pastSubmissionsReports/PastSubmissionReports.tsx:41]: This prob does not need to be global state but we can work it out later. It is ok for now

File: src/containers/pastSubmissionsSchools/PastSubmissionsSchools.tsx [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+  const dispatch = useDispatch();
+  const history = useHistory();
+
+  const availableSchools: AsyncRequest<
+    PastSubmissionsSchoolsResponse,
+    any
+  > = useSelector(
+    (state: C4CState) =>
+      state.pastSubmissionSchoolsState.pastSubmissionsSchools,
+  );
```
Comment (maxsebso) [inline src/containers/pastSubmissionsSchools/PastSubmissionsSchools.tsx:34]: This as well prob

**Review by maxsebso — APPROVED** [review]
LGTM, just pointed out some places where global state is not needed, but it does not really matter at this point in our application.

**Review by wwohlers — APPROVED** [review]

**Pull request merged**